Ginger/Ginger/Environments/AppDataBasesPage.xaml.cs

```diff
@@ -73,6 +86,50 @@ private void grdMain_CellEditEnding(object sender, DataGridCellEditEndingEventAr
                     }                   
                 }
             }
+
+            if (e.Column.Header.ToString() == nameof(Database.Name))
+            {
+                Database selectedDB = (Database)grdAppDbs.CurrentItem;
+                if (selectedDB.Name != selectedDB.NameBeforeEdit)
+                {
+                    UpdateDatabaseNameChange(selectedDB);
+                }
+            }
+        }
+
+        public void UpdateDatabaseNameChange(Database db)
+        {
+            if (db == null)
+            {
+                return;
+            }
+
+            Reporter.ToStatus(eStatusMsgKey.SaveItem, null, db.Name, "Database");
+            try
+            {
+                ObservableList<BusinessFlow> allBF = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<BusinessFlow>();
+                foreach (BusinessFlow businessFlow in allBF)
+                {
+                    foreach (Activity activity in businessFlow.Activities)
+                    {
+                        List<IAct> dbActs = activity.Acts.Where(x => (x.GetType() == typeof(ActDBValidation)) == true).ToList();
+                        foreach (ActDBValidation action in dbActs)
+                        {
+                            Database.UpdateDatabaseNameChangeInItem(action, db.NameBeforeEdit, db.Name);
+                        }
+                    }
+                }
+
+                db.NameBeforeEdit = db.Name;
```

Comment:
Try if we can get cells original value and edited value from sender object in grdMain_CellEditEnding event 
Instead of creating a new field to keep name before edit